### ISSUE TITLE
Scan OOVPA's per corresponding section first

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/Cxbx-Reloaded#138]
Cxbx currently scans the entire xbe contents for OOVPA matches. Accuracy can be improved by first scanning only the (library specific) section corresponding to the (library specific) OOVPA table being scanned with. Only after that's done, scan the entire xbe for xrefs not yet found. This will prevent detecting functions too early in the wrong section, when instead the correct match is most probably located inside the corresponding section.

**Comment by jackchentwkh** [comment]
As mentioned by the other issue, in order to improve the accuracy of OOVPA scan, there are few more ways to do. here are my two cents.
1. To scan the OOVPA by section first ast stated by PatrickvL above.
2. To scan the OOVPA with the LIB version info in the XBE header.
3. To improve the OOVPA signature.

1 and 2 are mentioned already. As for 3, I am thinking to add one more bytes for each OOVPA signature. As for each LIB function, the code length is somehow a signature, too. So if we could check the code length of each function in the LIB file, and get the final instruction byte as part of the OOVPA signature (should be RET for most cases I guess), this should improve the OOVPA accuracy, and avoid the unwanted duplicate matches.

**Comment by PatrickvL** [comment]
To find out what bytes OOVPA's end with, the patterns zip from the Dxbx project could be used.
It's a good idea. Not that these bytes are so unique, but they indicate a maximum length per OOVPA, which can be used to prevent incorrect overlaps at some point.

Note, that this won't work very well for LTCG titles. We should probably ignore LTCG for now, and wait for LLE (which will probably make OOVPA's completely obsolete anyway).

**Comment by PatrickvL** [comment]
I've updated the zip archive in our docs folder, it contains the Dxbx patterns on a more accessable form

**Comment by PatrickvL** [comment]
Do note however that currently some OOVPA's can be used for multiple versions of a function, even when these end differently

**Comment by PatrickvL** [comment]
@jarupxx / @Radwolfie Which of those remarks are still to be done? If all is done, please close this issue. If some parts are still valid, either update this issue, or split it off into a separate issue (and close this one)

**Comment by RadWolfie** [comment]
It's all done.
